main.go

```diff
@@ -102,3 +105,61 @@ func (mh *messageHandler) hackerNewsHandler(s *discordgo.Session, m *discordgo.M
 		log.Errorf("Could not send message over channel: %v", err)
 	}
 }
+
+var data = [][]string{{"Line1", "Hello Readers of"}, {"Line2", "golangcode.com"}}
+
+func (mh *messageHandler) hyperlinkCompilerHandler(s *discordgo.Session, m *discordgo.MessageCreate) {
+	// Ignore all messages created by the bot itself.
+	if m.Author.ID == s.State.User.ID {
+		return
+	}
+	// Check for command prefix
+	commandPrefix := "!links"
+	if !strings.Contains(m.Content, commandPrefix) {
+		return
+	}
+	// Read last 100 channel messages (ChannelMessages hard limit 😢)
+	messages, err := s.ChannelMessages(m.ChannelID, 100, "", "", "")
+	links := make(map[string]bool)
+	messageCount := 0
+	linkCount := 0
```

Comment:
No necesitas este link count ya que el numero de links es simplemente len(links) al final de tu for loop